### PR TITLE
ios_facts update documentation for gather_subset option

### PR DESCRIPTION
**Pull request by u571kills**
##### SUMMARY
Makes it clearer what options are available for gather_subset option.

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
ios_facts

##### ANSIBLE VERSION
```
2.4.0.0
```

##### ADDITIONAL INFORMATION

### CODE REVIEW

File: lib/ansible/modules/network/ios/ios_facts.py [hunk 1]
```diff
@@ -46,6 +46,7 @@
         not be collected.
     required: false
     default: '!config'
+    choices: ['all', 'hardware', 'config', 'interfaces']
```
Comment (ganeshrn) [inline lib/ansible/modules/network/ios/ios_facts.py:49]: For the choices given here `!` can be prefixed to specify that a specific subset should not be collected (which is also mentioned in `gather_subset ` description). As `choices` is not updated in the spec as well [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/ios/ios_facts.py#L447), so I think this is not valid documentation.

Comment (u571kills) [reply lib/ansible/modules/network/ios/ios_facts.py:49]: Are you suggesting I list all variations with '!' as well?

['all', 'hardware', 'config', 'interfaces', '!all', '!hardware', '!config', '!interfaces']

**Comment by gundalow** [comment]
Hi, apologies that this fell of my radar.
We've been discussing this a bit internally and we think we've got a better way of documenting this. I'll raise a PR with my thoughts so you can review.

**Comment by acozine** [comment]
Related to #43901.

**Comment by samccann** [comment]
@gundalow Does this mean this is good to merge, or should I close it out?

**Comment by samccann** [comment]
Based on slack conversation, @gundalow hopes to make further changes in this PR.

**Comment by samccann** [comment]
Closing in favor of https://github.com/ansible/ansible/pull/49264

**Pull request closed without merging**